submissions/KostiantynOkhorzin/oop-exercise/index.js

```diff
@@ -0,0 +1,61 @@
+
+class Inhabitan {
+   constructor(species, name, gender, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+      this.keys = ['species', 'name', 'gender', 'saying'];
+   }
+
+   showProps() {
+      return ['species', 'name', 'gender', 'saying'].map(key => this[key]).join('; ');
+   }
+};
+
+class Human extends Inhabitan {
+   constructor(name, gender, saying, legs = 2, hands = 2) {
+      super('human', name, gender, saying);
+      this.legs = legs;
+      this.hands = hands;
+   }
+
+   connectsProps() {
+      const props = super.showProps().split('; ');
+      const propsStart = props.slice(0, props.indexOf('gender'));
+      const propsEnd = props.slice(props.indexOf('gender'));
+      return [...propsStart, this.legs, this.hands, ...propsEnd].join('; ');
```

Comment:
I see, we want to insert props among inherited. Ok. 
Since we anyway refer to a property that doesn't belong to a child class, we can switch over to "base class somehow deals with children props".
`connectProps` is absically a duplication of code with a minor difference.
We may call inherited methods with some properties unique to the child class. Also we can call inherited methods from overloaded method.
The goal is to make code DRYer.

Reply:
I see, you opted not to keep this logic.